cellmlmanip/model.py

```diff
@@ -825,7 +832,8 @@ def _create_new_deriv_variable_and_equation(self, eqn, derivative_variable):
         """
         # 1. create a new variable
         deriv_name = self._get_unique_name(derivative_variable.name + '_orig_deriv')
-        deriv_units = self.units.calculator.traverse(eqn.args[0])
+        # TODO REWRITE THIS WITHOUT USING PRIVATE PROPERTIES
```

Comment:
Why does the calculator need to be private?

Reply:
As far as I can tell it's only a class (instead of a method) because it's long and has a few nested functions (now implemented as hidden methods of the class).

From a user's point of view it will be a lot simpler if you can just say e.g. `UnitStore.get_derived_units(expression)` and don't need to know that there's an exta class because it made the code prettier